release-notes/9.0/supported-os.json

```diff
@@ -0,0 +1,277 @@
+                },
+                {
+                    "name": "Ubuntu",
+                    "lifecycle-policy": "https://wiki.ubuntu.com/Releases",
+                    "architectures": [
+                        "x64",
+                        "arm64",
+                        "arm"
+                    ],
+                    "supported-cycles": [
+                        "20.04",
+                        "22.04",
+                        "23.10",
+                        "24.04"
+                    ]
+                }
+            ]
+        },
+        {
+            "family": "Windows",
+            "distributions": [
+                {
+                    "name": "Nano Server",
+                    "lifecycle-policy": "https://learn.microsoft.com/windows-server/get-started/windows-server-release-info",
+                    "architectures": [
+                        "x64"
+                    ],
+                    "supported-cycles": [
+                        "Version 1809+"
+                    ]
+                },
+                {
+                    "name": "Windows 10",
+                    "lifecycle-policy": "https://support.microsoft.com/help/13853/windows-lifecycle-fact-sheet",
+                    "architectures": [
+                        "x64",
+                        "x86",
+                        "arm64"
+                    ],
+                    "supported-cycles": [
+                        "Version 1607+"
+                    ]
+                },
+                {
+                    "name": "Windows 11",
+                    "lifecycle-policy": "https://support.microsoft.com/help/13853/windows-lifecycle-fact-sheet",
+                    "architectures": [
+                        "x64",
+                        "x86",
+                        "arm64"
+                    ],
+                    "supported-cycles": [
+                        "Version 22000+"
+                    ]
+                },
+                {
+                    "name": "Windows Server",
+                    "lifecycle-policy": "https://learn.microsoft.com/windows-server/get-started/windows-server-release-info",
+                    "architectures": [
+                        "x64",
+                        "x86"
+                    ],
+                    "supported-cycles": [
+                        "2012+"
+                    ]
+                },
+                {
+                    "name": "Windows Server Core",
+                    "lifecycle-policy": "https://learn.microsoft.com/windows-server/get-started/windows-server-release-info",
+                    "architectures": [
+                        "x64",
+                        "x86"
+                    ],
+                    "supported-cycles": [
+                        "2012+"
+                    ]
+                }
+            ]
+        }
+    ],
+    "linux": {
+        "glibc": [
+            {
+                "architectures": [
```

Comment:
Is the architecture the key here? What I mean is, is the intent that for a given Linux distribution I use the architecture to find the source/version of glibc for that?

Reply:
This mostly informs on what architectures are supported. Since arm32 and x86 are not listed, those are not in support.

Reply:
Thats in place so that this section could be generated: https://github.com/dotnet/core/blob/main/release-notes/9.0/supported-os.md#libc-compatibility

The overall doc/JSON tells you which versions that Microsoft chooses to support. The libc information could be used by a tool to provide a decent educated guess on OS versions where .NET will likely work but isn't supported by Microsoft.